### PR TITLE
Update React Native code formatting

### PR DESCRIPTION
**Pull request by mironiasty**
## Description

Various RN documentation tweaks:
- fix formatting configuration
- hide components from menu
- refactor all examples to be 'copy - ready'

This is how updated example will look like:

Screenshot 2024-11-28 at 10 53 48

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "tabWidth": 2
```
Comment (mironiasty) [inline .prettierrc:2]: I've added this as sometimes vscode does not use prettier formatter if config file is missing

File: docs/react-native/_components/configure-permissions.mdx
Comment (mironiasty) [inline docs/react-native/_components/configure-permissions.mdx:1]: This rename was required, as right now this files are visible as separate documentation: https://docs.fishjam.io/react-native/components/configure-permissions

File: .prettierignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 build/*
-node_modules/*
\ No newline at end of file
+node_modules/*
+.yarn/*
```
Comment (maksg) [inline .prettierignore:3]: New line at the end, pretty please 🥺

File: docs/react-native/connecting.mdx [hunk 3]
```diff
@@ -45,22 +45,43 @@ follow our [server setup instructions](/production/server).
...
+import { Button } from "react-native";
 import { useConnection } from "@fishjam-cloud/react-native-client";
 
-const { joinRoom } = useConnection();
+export function JoinRoomButton() {
+  // highlight-next-line
+  const { joinRoom } = useConnection();
+
+  const onPressJoin = useCallback(async () => {
+    const { fishjamUrl, peerToken } = yourFunction();
```
[the remark this answers is not in the document]
Comment (mironiasty) [reply docs/react-native/connecting.mdx]: I've implemented function to get data from sandbox as an example

File: docs/react-native/quick-setup.mdx [hunk 4]
```diff
@@ -63,26 +63,34 @@ Keep in mind that this won't work on iOS Simulator, as Simulator can't access th
...
+import { useCallback } from "react";
+import { Button } from "react-native";
+import { useCamera, useConnection } from "@fishjam-cloud/react-native-client";
+
+export function StartStreamingButton({
   roomName,
   userName,
 }: {
   roomName: string;
   userName: string;
```
[the remark this answers is not in the document]
Comment (mironiasty) [reply docs/react-native/quick-setup.mdx]: Updated everywhere

**Review by MiloszFilimowski — APPROVED** [review]

**Pull request merged**